thirdeye-persistence/src/main/java/ai/startree/thirdeye/datalayer/bao/SubscriptionGroupManagerImpl.java

```diff
@@ -16,15 +16,36 @@
 import ai.startree.thirdeye.datalayer.dao.GenericPojoDao;
 import ai.startree.thirdeye.spi.datalayer.bao.SubscriptionGroupManager;
 import ai.startree.thirdeye.spi.datalayer.dto.SubscriptionGroupDTO;
+import com.codahale.metrics.CachedGauge;
+import com.codahale.metrics.MetricRegistry;
 import com.google.inject.Inject;
 import com.google.inject.Singleton;
+import java.util.concurrent.TimeUnit;
 
 @Singleton
 public class SubscriptionGroupManagerImpl extends
     AbstractManagerImpl<SubscriptionGroupDTO> implements SubscriptionGroupManager {
 
   @Inject
-  public SubscriptionGroupManagerImpl(GenericPojoDao genericPojoDao) {
+  public SubscriptionGroupManagerImpl(final GenericPojoDao genericPojoDao,
+      final MetricRegistry metricRegistry) {
     super(SubscriptionGroupDTO.class, genericPojoDao);
+    metricRegistry.register("subscriptionsCountTotal",
+        new CachedGauge<Integer>(15, TimeUnit.MINUTES) {
+      @Override
+      public Integer loadValue() {
+        return findAll().stream()
+            .filter(SubscriptionGroupManagerImpl::isPairPresent)
```

Comment:
Instead, just chain the filters. that would be just as readable.

Reply:
Oh I thought the chaining will make it hard to understand why are we having the filters😂 hence I created a separate method so that the filter is more readable. Totally defeated the cause here. Thanks for the feedback, will chain the filters👍